targets/TARGET_ARM_SSG/TARGET_MUSCA_A1/partition/flash_layout.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2018 Arm Limited. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __FLASH_LAYOUT_H__
+#define __FLASH_LAYOUT_H__
+
+/* Flash layout on Musca with BL2:
+ * 0x0020_0000 BL2 - MCUBoot(128 KB)
+ * 0x0022_0000 Flash_area_image_0(1 MB)
+ *    0x0022_0000 Secure     image primary
+ *    0x002A_0000 Non-secure image primary
+ * 0x0032_0000 Flash_area_image_1(1 MB)
+ *    0x0032_0000 Secure     image secondary
+ *    0x003A_0000 Non-secure image secondary
+ * 0x0042_0000 Secure Storage Area(0.02 MB)
+ * 0x0042_5000 NV counters area(16 Bytes)
+ * 0x0042_5010 Unused
+ */
+
+/* Code SRAM layout on Musca (with BL2, which is mandatory) after the newest
+ * image has been copied to SRAM:
+ * 0x0000_0000 BL2 - MCUBoot(128 KB)
+ * 0x0002_0000 Flash_area_newest_image(1 MB)
+ *    0x0002_0000 Secure     image primary
+ *    0x000A_0000 Non-secure image primary
+ * 0x0012_00000 Unused
+ */
+
+/* This header file is included from linker scatter file as well, where only a
+ * limited C constructs are allowed. Therefore it is not possible to include
+ * here the platform_retarget.h to access flash related defines. To resolve this
+ * some of the values are redefined here with different names, these are marked
+ * with comment.
+ */
+
+/* The size of a partition. This should be large enough to contain a S or NS
+ * sw binary. Each FLASH_AREA_IMAGE contains two partitions. See Flash layout
+ * above.
+ */
+#define FLASH_PARTITION_SIZE            (0x80000) /* 512KB */
+
```

Comment:
Shouldn't we use value from targets.json inherited from PSA_Target:
"overrides": {
"secure-rom-start": "xx",
"secure-rom-size": "xx",
"non-secure-rom-start": "xx",
"non-secure-rom-size": "xx",
"secure-ram-start": "xx",
"secure-ram-size": "xx",
"non-secure-ram-start": "xx",
"non-secure-ram-size": "xx",
"shared-ram-start": "xx",
"shared-ram-size": "xx"
}

Reply:
Yes, we are going to align the partition files with the configuration.